synapse/api/errors.py

```diff
@@ -510,7 +511,12 @@ def __init__(
         retry_after_ms: Optional[int] = None,
         errcode: str = Codes.LIMIT_EXCEEDED,
     ):
-        super().__init__(code, msg, errcode)
+        headers = (
+            None
+            if retry_after_ms is None
+            else {"Retry-After": str(math.ceil(retry_after_ms / 1000))}
```

Comment:
I find this odd, but seems to be what the MSC says.

Reply:
Yep, just find it odd instead of `int(retry_after_ms)`, which is much simpler. 🤷

Reply:
Oh, because if you got a `retry_after_ms` of 5100ms, and you round *down* to `5s` then your client may retry sooner? The ceil gives you safety that you won't exceed the *real* retry time.

Reply:
Right, that makes sense. 👍